src/Forms/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -952,41 +953,56 @@ void ConfigureCarouselPage(CarouselPage carouselPage, string segment)
             CarouselPageSelectTab(carouselPage, parameters);
         }
 
-        private static void SelectPageTab(Page page, INavigationParameters parameters)
+        private async Task SelectPageTab(Page page, INavigationParameters parameters)
         {
             if (page is TabbedPage tabbedPage)
             {
-                TabbedPageSelectTab(tabbedPage, parameters);
+                await TabbedPageSelectTab(tabbedPage, parameters);
             }
             else if (page is CarouselPage carouselPage)
             {
                 CarouselPageSelectTab(carouselPage, parameters);
             }
         }
 
-        private static void TabbedPageSelectTab(TabbedPage tabbedPage, INavigationParameters parameters)
+        private async Task TabbedPageSelectTab(TabbedPage tabbedPage, INavigationParameters parameters)
         {
             var selectedTab = parameters?.GetValue<string>(KnownNavigationParameters.SelectedTab);
             if (!string.IsNullOrWhiteSpace(selectedTab))
             {
-                var selectedTabType = PageNavigationRegistry.GetPageType(UriParsingHelper.GetSegmentName(selectedTab));
+                string segment;
+                Queue<string> tabSegements = null;
+                if (selectedTab.Contains('|'))
+                {
+                    tabSegements = new Queue<string>(selectedTab.Split('|'));
+                    segment = tabSegements.Dequeue();
+                }
+                else
+                {
+                    segment = selectedTab;
+                }
+
+                var selectedTabType =
+                    PageNavigationRegistry.GetPageType(UriParsingHelper.GetSegmentName(segment));
```

Comment:
this currently isn't iterating like it needs to so if you have `NavigationPage|ViewA|ViewB|ViewC` you only end up with `NavigationPage|ViewA`